modules/kubeconfig/main.tf

```diff
@@ -19,11 +19,24 @@ data "template_file" "kubeconfig_provider" {
   }
 }
 
+# 1.24 change: Create service account secret
+resource "kubernetes_secret" "sa_secret" {
+  metadata {
```

Comment:
There should be a guard on this code base. Always list the data element before the resource. You'll need a guard like this:

`count = var.create_stataic_kubeconfig ? 0 : 1` 

for this block.

We either use the data source or create our own resource.

Reply:
I'll add that.

Reply:
Added in the requested change.

Reply:
The logic above is correct. The data source and the resource for secret are not either or. Previously the data was reading the secret from the service account's default secret however starting K8s v1.24.0 default secret is no longer being generated for the service account and hence the data can't get any secret. So the solution to this was to create a new resource kubernetes_secret which provides the secret to the data source.
Discussed this with Thomas, the usage of data/resource pair and the referencing syntax in the current code is different. In the latest change I have updated the logic to only reference the resource and removed the data source for the secret.